runtime/src/accounts_db.rs

```diff
@@ -53,11 +55,33 @@ use std::{
 use tempfile::TempDir;
 
 const PAGE_SIZE: u64 = 4 * 1024;
+const MAX_RECYCLE_STORES: usize = 1000;
+const STORE_META_OVERHEAD: usize = 256;
+const MAX_CACHE_SLOTS: usize = 200;
```

Comment:
btw, why this value is chosen?

Reply:
```
$ du -shc $(ls -tr target/mainnet-beta72/accounts/*59546645*)
...
3.1M    target/mainnet-beta72/accounts/59546645.896465646
1.1M    target/mainnet-beta72/accounts/59546645.896465648
3.1M    target/mainnet-beta72/accounts/59546645.896465625
3.2M    target/mainnet-beta72/accounts/59546645.896465629
...
170M    total

$ pry
[2] pry(main)> 170 * 200
=> 34000
```

So, this could generally require heap of 34G? How about reducing this a bit or making the eviction as bytes-written based one?

Reply:
So ideally I think we cache as much as possible so we can do the smart pruning later
to save disk ops.

I figured this `MAX_CACHE_SLOTS` should be roughly the snapshot interval since that's when we *need* the information on disk. So currently the cache will be forced flush every 100 roots before the  snapshot is taken. Given that there should be at least 32 unrooted slots on top of that, this is a lower bound of 132 or so slots when the snapshot is about to be taken. I figured if there are some skipped blocks in there, maybe skip rate of 33%, that's about 166 slots, so I rounded up to 200.

Currently the cache is sitting at about 5GB for 100 slots, checking this metric `accounts_cache_size` here: https://metrics.solana.com:8888/sources/3/chronograf/data-explorer?query=SELECT%20mean%28%22total_unique_writes_size%22%29%20AS%20%22mean_total_unique_writes_size%22%2C%20mean%28%22num_slots%22%29%20AS%20%22mean_num_slots%22%20FROM%20%22mainnet-beta%22.%22autogen%22.%22accounts_cache_size%22%20WHERE%20time%20%3E%20now%28%29%20-%2012h%20AND%20%22host_id%22%3D%27HQUTs2ntTrmqJPVBnFJ9DN7uLHMHeooEy8XDTPkrBwE1%27%20GROUP%20BY%20time%28%3Ainterval%3A%29%20FILL%28null%29

So the upper bound should be around 10GB for now, I think bytes-based eviction might not be a bad idea, but we can do that in a follow-up PR

Reply:
> I figured this `MAX_CACHE_SLOTS` should be roughly the snapshot interval since that's when we _need_ the information on disk. So currently the cache will be forced flush every ...

got it. makes sense.

> lower than your calculation because fewer duplicate writes

oh yeah, I forgot about the duplicate intra-slot writes. ,,